clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```diff
@@ -930,6 +931,120 @@ ElectLeadersResult electLeaders(
         Set<TopicPartition> partitions,
         ElectLeadersOptions options);
 
+
+    /**
+     * Change the reassignments for one or more partitions.
+     * Providing an empty Optional (e.g via {@link Optional#empty()}) will <bold>cancel</bold> the reassignment for the associated partition.
+     *
+     * This is a convenience method for {@link #alterPartitionReassignments(Map, AlterPartitionReassignmentsOptions)}
+     * with default options.  See the overload for more details.
+     */
+    default AlterPartitionReassignmentsResult alterPartitionReassignments(
+        Map<TopicPartition, Optional<NewPartitionReassignment>> reassignments) {
+        return alterPartitionReassignments(reassignments, new AlterPartitionReassignmentsOptions());
+    }
+
+    /**
+     * Change the reassignments for one or more partitions.
+     * Providing an empty Optional (e.g via {@link Optional#empty()}) will <bold>cancel</bold> the reassignment for the associated partition.
+     *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on the futures obtained from
+     * the returned {@code AlterPartitionReassignmentsResult}:</p>
+     * <ul>
+     *   <li>{@link org.apache.kafka.common.errors.ClusterAuthorizationException}
+     *   If the authenticated user didn't have alter access to the cluster.</li>
+     *   <li>{@link org.apache.kafka.common.errors.UnknownTopicOrPartitionException}
+     *   If the topic or partition does not exist within the cluster.</li>
+     *   <li>{@link org.apache.kafka.common.errors.TimeoutException}
+     *   if the request timed out before the controller could record the new assignments.</li>
+     *   <li>{@link org.apache.kafka.common.errors.InvalidReplicaAssignmentException}
```

Comment:
I think we should opt out of this excepiton. In my controller changes, I haven't implemented it -- mainly because it's tricky to know when a replica is intermittently offline, as the znode and controller memory cache store only the online brokers.

Reply:
Can you think of a scenario where we'd want to create a reassignment to a broker that isn't up?  It seems like most (all?) of the time, this would be a mistake.

Another reason why the assignment could be invalid is if the same broker ID appeared more than once (reassignment to 1, 1, 1, etc.)

Reply:
Come to think of it... I can't. I think it makes sense to have as long as we do the validation on API calls directly